cosmicfishpie/cosmology/cosmology.py

```diff
@@ -135,33 +169,122 @@ def set_cosmicfish_defaults(self):
             )
 
         # Set default value for Neff if it is not found in cosmopars
-        self.cosmopars["Neff"] = self.cosmopars.get("Neff", self.cosmopars.get("N_eff", 3.046))
+        # self.cosmopars["Neff"] = self.cosmopars.get("Neff", self.cosmopars.get("N_eff", 3.046))
 
         # Set default value for gamma, if it is not found in cosmopars
         # gamma is not used in many places, therefore not needed to add back in cosmopars
         self.gamma = self.cosmopars.get("gamma", 0.545)
 
-    def camb_setparams(self, cosmopars, camb):
+    @staticmethod
+    def print_cosmo_params(cosmopars, feedback=1, text="---Cosmo pars---"):
+        """
+        Print cosmological parameters.
+
+        Parameters:
+        -----------
+        cosmopars : dict
+            Dictionary of cosmological parameters to print.
+        feedback : int, optional
+            Feedback level determining whether to print (default is 1).
+        text : str, optional
+            Header text for the parameter list (default is "---Cosmo pars---").
+        """
+        if feedback > 2:
+            print("")
+            print(text)
+            for key in cosmopars:
+                print(key + "=" + str(cosmopars[key]))
+
+    @staticmethod
+    def f_deriv(D_growth_zk, z_array, k_array, k_fix=False, fixed_k=1e-3):
         """
-        Sets the parameters for the CAMB (Code for Anisotropies in the Microwave Background) computation.
+        Calculate the growth rate f(z,k).
 
         Parameters:
-            cosmopars (dict): A dictionary containing the cosmological parameters.
-            camb: The CAMB object.
+        -----------
+        D_growth_zk : callable
+            Function that returns the growth factor D(z,k).
+        z_array : array_like
+            Array of redshift values.
+        k_array : array_like
+            Array of wavenumbers.
+        k_fix : bool, optional
+            If True, use a fixed k value (default is False).
+        fixed_k : float, optional
+            Fixed k value to use if k_fix is True (default is 1e-3).
 
         Returns:
-            None
+        --------
+        tuple
+            Growth rate f(z,k) and the corresponding z array.
+        """
+        if k_fix:
+            k_array = np.full((len(k_array)), fixed_k)
+        ## Generates interpolators D(z) for varying k values
+        D_z = np.array([UnivariateSpline(z_array, D_growth_zk(z_array, kk), s=0) for kk in k_array])
```

Comment:
I agree, but since we validated CF in this way, I am afraid of moving from Univariate to Bivariate and everything that interpolators do. Also, not sure about the .derivative() method. Can we handle this in a different MR?

Reply:
Sure